cmake/onnxruntime_csharp.cmake

```diff
@@ -10,11 +10,35 @@ else()
   set (CSHARP_DEPENDS onnxruntime)
 endif()
 
+if (onnxruntime_USE_CUDA)
+  STRING(APPEND CSHARP_PREPROCESSOR_DEFINES "USE_CUDA,")
```

Comment:
Indeed you may use the 'ONNXRUNTIME_PROVIDER_NAMES' cmake variable to replace the long list of if(..) ...endif()